packages/core-flows/src/handlers/promotion/update-campaigns.ts

```diff
@@ -0,0 +1,36 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IPromotionModuleService, UpdateCampaignDTO } from "@medusajs/types"
+import { getSelectsAndRelationsFromObjectArray } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+export const updateCampaignsStep = createStep(
+  "update-campaigns",
+  async (data: UpdateCampaignDTO[], { container }) => {
+    const promotionModule = container.resolve<IPromotionModuleService>(
+      ModuleRegistrationName.PROMOTION
+    )
+
+    const { selects, relations } = getSelectsAndRelationsFromObjectArray(data)
+    const dataBeforeUpdate = await promotionModule.listCampaigns(
+      { id: data.map((d) => d.id) },
+      { relations, select: selects }
+    )
+
+    const updatedCampaigns = await promotionModule.updateCampaigns(data)
+
+    return new StepResponse(updatedCampaigns, dataBeforeUpdate)
+  },
+  async (dataBeforeUpdate, { container }) => {
+    if (!dataBeforeUpdate) {
+      return
+    }
+
+    const promotionModule = container.resolve<IPromotionModuleService>(
+      ModuleRegistrationName.PROMOTION
+    )
+
+    // TODO: This still requires some sanitation of data and transformation of
```

Comment:
question: Do you want to include this now?

Reply:
Not required for now as it will never end up in this place anyway in its current form. I've created a ticket to handle it next week.